src/libstd/sys/unix/stack_overflow.rs

```diff
@@ -134,17 +134,42 @@ mod imp {
         Handler { _data: MAIN_ALTSTACK };
     }
 
-    pub unsafe fn make_handler() -> Handler {
-        let alt_stack = mmap(ptr::null_mut(),
-                             SIGSTKSZ,
-                             PROT_READ | PROT_WRITE,
-                             MAP_PRIVATE | MAP_ANON,
-                             -1,
-                             0);
-        if alt_stack == MAP_FAILED {
+    #[cfg(any(target_os = "linux",
+              target_os = "macos",
+              target_os = "bitrig",
+              target_os = "netbsd",
+              target_os = "openbsd"))]
+    unsafe fn get_stack() -> *mut libc::c_void {
```

Comment:
they are the same except the cast at the end.  we don't have conditional compilation for general statements yet and I couldn't think of any other way to make this compile.  on dragonflybsd and freebsd, the stack.ss_sp field is a *mut i8 type and on the others it is a *mut libc::c_void.  Doing a runtime cfg!() doesn't work because one branch always fails to compile due to the warn -> error compilation flags.

I asked in #rust today if there was a better way to do this and nobody could think of a better way.  @alexcrichton do you know of a better way to do this?